pkg/cache/v3/status.go

```diff
@@ -155,3 +159,26 @@ func (info *statusInfo) setDeltaResponseWatch(id int64, drw DeltaResponseWatch)
 	defer info.mu.Unlock()
 	info.deltaWatches[id] = drw
 }
+
+// orderResponseWatches will track a list of watch keys and order them if
+// true is passed.
+func (info *statusInfo) orderResponseWatches(order bool) {
+	info.orderedWatches = make(keys, len(info.watches))
+
+	// This runs in O(n) which could become problematic when we have an extremely high watch count.
```

Comment:
You're right the sort is really fast but building that indexed ordered list is worse case scenario `O(len(info.watches))`. Luckily inserting into a map is `O(1)`. Honestly I can probably delete this comment because this operation really isn't that expensive.